### ISSUE TITLE
Another "classification" highlighting issue

### ISSUE THREAD

**Issue by borama** [Shopify/ruby-lsp#2302]
### Description

(VSCode 1.91.0 , Ruby LSP 0.7.12)

Related to #2269, I recently noticed another issue with words containing `class` being highlighted as keywords. This time, it is words _ending_ with `class`. 

I [noticed](https://github.com/borama/vscode-ruby-slim/issues/2) it in a Slim template similar to this:

```slim
div class=wrapper_classes data-something="value"

div class=wrapper_class data-something="value"
```

Supposing `wrapper_classes` and `wrapper_class` are some ruby methods, the first example is highlighted correctly while the second is messed up:

![obrazek](https://github.com/user-attachments/assets/e711d31b-6e34-41fa-93d1-257c09b30ffe)

Adding a word boundary check to the beginning of the [class regexp](https://github.com/Shopify/ruby-lsp/blob/main/vscode/grammars/ruby.cson.json#L68) in the grammar file seems to fix it:

```diff
- "match": "(class)\\s+(([a-zA-Z0-9_]+)((::)[a-zA-Z0-9_]+)*)\\s*((<)\\s*(([a-zA-Z0-9_]+)((::)[a-zA-Z0-9_]+)*))?",
+ "match": "\b(class)\\s+(([a-zA-Z0-9_]+)((::)[a-zA-Z0-9_]+)*)\\s*((<)\\s*(([a-zA-Z0-9_]+)((::)[a-zA-Z0-9_]+)*))?",
```

I can add a PR if you like. Thanks!

**Comment by andyw8** [comment]
@borama when you are editing this file, does VS Code show `{ } Ruby` in the footer?

**Comment by borama** [comment]
@andyw8 No, it says "Slim" as the file is a [Slim template](https://github.com/slim-template/slim) but, in Slim notation, everything after the equal sign in an attribute is [embedded ruby](https://github.com/slim-template/slim?tab=readme-ov-file#ruby-attributes) and that's where I think the ruby LSP comes into play and that's also where the `class` keyword gets matched via the regexp. See the Editor tokens inspection:

![image](https://github.com/user-attachments/assets/93b5537b-451b-4e9d-a352-805c45e18e9e)

BTW I also managed to replicate the same problem in a [HAML template](https://haml.info/docs/yardoc/file.REFERENCE.html): suppose I have a `some_class` helper method available, then the following HAML template renders correctly:

```ruby
def some_class(item)
  "some_class #{item}"
end

hi = "hi!"
engine = Haml::Template.new { "%p #{some_class hi}" }
engine.render

# outputs:
=> "<p>some_class hi!</p>\n"
```

but when I write the same HAML code in a HAML template file and have one of the HAML extensions installed in VS Code, then it does not get highlighted properly:

![image](https://github.com/user-attachments/assets/cc08e8c6-40ff-4fbd-af21-40395472ebea)

Again, it is embedded ruby inside the parens and due to the `class` regexp _not_ being properly left-bounded, it highlights the part of the helper class as a ruby keyword.

**Comment by borama** [comment]
Oh I can actually replicate this even in a ruby file, at least for a moment. When I paste the following code to a bigger ruby file (that takes VS Code some time to parse), the incorrect highlighting is briefly visible before the [semantic highlighting](https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide) takes place and corrects it:

[Záznam obrazovky z 2024-07-26 00-10-52.webm](https://github.com/user-attachments/assets/8d41fd44-f713-4fe6-90f9-f13d036916c9)

**Comment by andyw8** [comment]
Which Slim extension are you using, [this one](https://marketplace.visualstudio.com/items?itemName=sianglim.slim)?

**Comment by borama** [comment]
No, I'm using [my own extension](https://marketplace.visualstudio.com/items?itemName=borama.ruby-slim) that I made recently :) but both of them behave similarly and fail in slightly different situations, based on what the extension recognizes as ruby (the older one e.g. does not highlight ruby in HTML attributes).

**Comment by andyw8** [comment]
Ah, I see. I was able to reproduce the issue with your extension.

This should solve it, if you want to give the branch a try:

https://github.com/Shopify/ruby-lsp/pull/2371

Note that you'll need to run via the Extension Development Host so it uses the updated grammar.

**Comment by andyw8** [comment]
Another way to illustrate the problem is with a Ruby heredoc string, e.g.:

<img width="167" alt="Screenshot 2024-07-26 at 9 46 34 AM" src="https://github.com/user-attachments/assets/aa823994-47f5-47e7-aa9f-c45296dab44d">

**Comment by borama** [comment]
Ah, indeed, this seems perfect, thank you!

**Comment by borama** [comment]
Fixed in #2371 :pray:
